test/stream.spec.js

```diff
@@ -0,0 +1,194 @@
+/* eslint-env mocha */
+'use strict'
+
+const chai = require('chai')
+const dirtyChai = require('dirty-chai')
+const expect = chai.expect
+chai.use(dirtyChai)
+
+const { decode } = require('length-prefixed-stream')
+const { createDaemon } = require('libp2p-daemon/src/daemon')
+const Client = require('../src')
+const { StreamInfo } = require('libp2p-daemon/src/protocol')
+
+const { ends } = require('../src/util/iterator')
+const { getSockPath } = require('./utils')
+const defaultSock = getSockPath('/tmp/p2pd.sock')
+
+describe('daemon stream client', function () {
+  this.timeout(50e3)
+
+  const sock2 = getSockPath('/tmp/p2pd-2.sock')
+  let daemonA
+  let daemonB
+  let clientA
+  let clientB
+
+  const daemonOpts = (sock) => ({
+    quiet: false,
+    q: false,
+    bootstrap: false,
+    hostAddrs: '/ip4/0.0.0.0/tcp/0,/ip4/0.0.0.0/tcp/0/ws',
+    b: false,
+    dht: false,
+    dhtClient: false,
+    connMgr: false,
+    sock: sock || defaultSock,
+    id: '',
+    bootstrapPeers: ''
+  })
+
+  before(function () {
+    return Promise.all([
+      createDaemon(daemonOpts()),
+      createDaemon(daemonOpts(sock2))
+    ]).then((res) => {
+      daemonA = res[0]
+      daemonB = res[1]
+
+      return Promise.all([
+        daemonA.start(),
+        daemonB.start()
+      ])
+    })
+  })
+
+  after(async function () {
+    this.timeout(100e3)
```

Comment:
Is this timeout necessary? It looks like the tests are passing pretty quickly.

Reply:
it should not be! I added it for debugging purposes and forgot to revert. Removed now, let's wait for CI